### PR TITLE
Move parameter registration to translation units

### PR DESCRIPTION
**Pull request by akva2**
To avoid compiling the code in every simulator object.

### CODE REVIEW

**Comment by akva2** [comment]
jenkins build this please

File: opm/simulators/flow/BlackoilModelParameters.cpp [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+    local_well_solver_control_switching_ = Parameters::Get<Parameters::LocalWellSolveControlSwitching>();
+    use_implicit_ipr_ = Parameters::Get<Parameters::UseImplicitIpr>();
+    nonlinear_solver_ = Parameters::Get<Parameters::NonlinearSolver>();
+    const auto approach = Parameters::Get<Parameters::LocalSolveApproach>();
+    if (approach == "jacobi") {
+        local_solve_approach_ = DomainSolveApproach::Jacobi;
+    } else if (approach == "gauss-seidel") {
+        local_solve_approach_ = DomainSolveApproach::GaussSeidel;
+    } else {
+        throw std::runtime_error("Invalid domain solver approach '" + approach + "' specified.");
```
Comment (bska) [inline opm/simulators/flow/BlackoilModelParameters.cpp:77]: I'm sure we get it through some transitive include statement, but we should explicitly
```C++
#include <stdexcept>
```
to bring a declaration of the `std::runtime_error` type into scope.

**Review by bska — COMMENTED** [review]
Save for one potential missing `#include` statement this looks good to me.

File: opm/simulators/flow/BlackoilModelParameters.cpp (hunk 1 above)
Comment (akva2) [reply opm/simulators/flow/BlackoilModelParameters.cpp:78]: sure, just an oversight..

**Comment by akva2** [comment]
jenkins build this please

**Pull request merged**